### PR TITLE
Feature: Adding method to update order status history

### PR DESCRIPTION
**Pull request by eapg**
* Adding method to update order status history in repository
* Adding test for this method
* updating order status history: order to order_id

### CODE REVIEW

File: src/core/engine/app_engine_processor.py [hunk 1]
```diff
@@ -3,13 +3,15 @@
...
+from src.core.ioc import get_ioc_instance
 
 
 class AppEngineProcessor:
     def __init__(self):
+        ioc_instance = get_ioc_instance()
         kitchen_simulator_config = AppProcessorConfig("kitchen_simulator_test", 10)
         kitchen_simulator = KitchenSimulator(kitchen_simulator_config)
-        self.app_context = AppEngineProcessorContext(processors=[kitchen_simulator])
+        self.app_context = AppEngineProcessorContext(processors=[kitchen_simulator], ioc_instance=ioc_instance)
```
Comment (reynoldmorel) [inline src/core/engine/app_engine_processor.py:14]: Is not necessary to add the ioc instance since it is a singleton instance for the whole application

**Review by reynoldmorel — CHANGES_REQUESTED** [review]

File: src/core/engine/app_engine_processor.py (hunk 1 above)
Comment (eapg) [reply src/core/engine/app_engine_processor.py]: done, i got the ioc instance in the kitchen simulator, as is the same instance in the whole application

File: src/core/engine/processors/kitchen_simulator.py [hunk 2]
```diff
@@ -1,13 +1,124 @@
...
+        self, order=None, from_status=None, from_time=None, to_status=None, to_time=None
+    ):
+        new_order_status_history = OrderStatusHistory()
+        new_order_status_history.order = order
+        new_order_status_history.from_status = from_status
+        new_order_status_history.from_time = from_time
+        new_order_status_history.to_status = to_status
+        new_order_status_history.to_time = to_time
+        return new_order_status_history
+
```
Comment (reynoldmorel) [inline src/core/engine/processors/kitchen_simulator.py:33]: Could you handle different statses using differen processes?

Example:

process_new_orders()
process_placed_orders()
cleaned_ordes_queue()

**Review by reynoldmorel — CHANGES_REQUESTED** [review]

File: src/tests/lib/repositories/impl/order_status_history_repository_impl_test.py [hunk 3]
```diff
@@ -130,3 +129,21 @@ def test_get_by_order_id_successfully(self):
...
+        )
+        order_1.status = OrderStatus.ORDER_PLACED
+        order_status_history_repository.set_next_status_history_by_order_id(
+            order_1.id, order_1.status
+        )
+        order_status_histories = order_status_history_repository.get_all()
+        self.assertEqual(
+            order_status_histories[0].to_time, order_status_histories[1].from_time
+        )
+        self.assertEqual(
```
Comment (reynoldmorel) [inline src/tests/lib/repositories/impl/order_status_history_repository_impl_test.py:147]: Could you assert get last status is being called?

**Review by reynoldmorel — COMMENTED** [review]
Also please remember to use the same pattern for relationships, I highly recommend to change all the existing to follow this pattern in other entities, I mean to use `[entity_name]_id`. Other than that this PR looks good.

If you are going to do that refactor, do it in another pr.

**Review by reynoldmorel — APPROVED** [review]

**Pull request merged**